test/jdk/java/net/httpclient/HttpResponseConnectionLabelTest.java

```diff
@@ -0,0 +1,446 @@
+
+            // Register the request handler
+            String serverId = "" + SERVER_COUNTER.getAndIncrement();
+            CountDownLatch[] serverResponseLatchRef = new CountDownLatch[1];
+            server.addHandler(
+                    (exchange) -> {
+                        String responseBody = "" + SERVER_RESPONSE_COUNTER.getAndIncrement();
+                        String connectionKey = exchange.getConnectionKey();
+                        try (exchange) {
+
+                            // Participate in the latch count down
+                            if (serverResponseLatchRef[0] != null) {
+                                serverResponseLatchRef[0].countDown();
+                                LOGGER.log(
+                                        "Server[%s] is waiting for the latch... (connectionKey=%s, responseBody=%s)",
+                                        serverId, connectionKey, responseBody);
+                                serverResponseLatchRef[0].await();
+                            }
+
+                            // Write the response
+                            LOGGER.log(
+                                    "Server[%s] is responding... (connectionKey=%s, responseBody=%s)",
+                                    serverId, connectionKey, responseBody);
+                            exchange.getResponseHeaders().addHeader(CONNECTION_KEY_HEADER_NAME, connectionKey);
+                            exchange.getResponseHeaders().addHeader(SERVER_ID_HEADER_NAME, serverId);
+                            byte[] responseBodyBytes = responseBody.getBytes(CHARSET);
+                            exchange.sendResponseHeaders(200, responseBodyBytes.length);
+                            exchange.getResponseBody().write(responseBodyBytes);
+
+                        } catch (IOException ioe) {
+                            synchronized (ServerRequestPair.class) {
+                                LOGGER.log(
+                                        "Server[%s] has failed! (connectionKey=%s, responseBody=%s)",
+                                        serverId, connectionKey, responseBody);
+                                ioe.printStackTrace(System.out);
+                            }
+                        } catch (InterruptedException _) {
+                            Thread.currentThread().interrupt(); // Restore the interrupt
+                        }
+                    },
+                    handlerPath);
+
+            // Create the client and the request
+            HttpRequest request = HttpRequest.newBuilder(requestUri).version(version).build();
+
+            // Create the pair
+            return new ServerRequestPair(server, request, secure, serverResponseLatchRef);
+
+        }
+
+        private static HttpTestServer createServer(Version version, SSLContext sslContext) {
+            try {
+                return HttpTestServer.create(version, sslContext, ForkJoinPool.commonPool());
```

Comment:
The HTTP/2 test server in the JDK shuts down the `Executor` that is passed to that server (like this one), when the server is stopped. I had a look at the specification and implementation of `ForkJoinPool.commonPool()` - it is specified to not close/shutdown the common pool. So I think it is OK to use the `commonPool()` as the `Executor` for these servers.

Although, since this test is agentvm test, we never known what state the commonPool() executor is in due to some previously run test(s). Maybe we should consider having a test specific Executor in this test, like we do in some other tests within the httpclient tests.

Reply:
Another possibility is to run in /othervm mode - then we don't care...

Reply:
Yes, I think othervm would be fine. In fact, I was adding a review comment in another context, where I suspect we might be forced to use the othervm mode anyway.

Reply:
Switched to `othervm` in cc70d926dbc867ecf8f3390c76ccb02888c8e8d8.

> I was adding a review comment in another context, where I suspect we might be forced to use the othervm mode anyway.

@jaikiran, curious, where is it?

Reply:
Hello Volkan, this one https://github.com/openjdk/jdk/pull/24154#discussion_r2028999507